qa/L0_socket/client.py

```diff
@@ -0,0 +1,139 @@
+    input0_data = np.expand_dims(input0_data, axis=0)
+    input1_data = np.ones(shape=(1, 16), dtype=np.int32)
+
+    inputs[0].set_data_from_numpy(input0_data)
+    inputs[1].set_data_from_numpy(input1_data)
+
+    outputs.append(client_util.InferRequestedOutput('OUTPUT0'))
+    outputs.append(client_util.InferRequestedOutput('OUTPUT1'))
+
+    results = []
+    for i in range(request_count):
+        results.append(client.infer(model_name, inputs, outputs=outputs))
+
+    for result in results:
+        output0_data = result.as_numpy('OUTPUT0')
+        output1_data = result.as_numpy('OUTPUT1')
+        for i in range(16):
+            print(str(input0_data[0][i]) + " + " + str(input1_data[0][i]) +
+                  " = " + str(output0_data[0][i]))
+            print(str(input0_data[0][i]) + " - " + str(input1_data[0][i]) +
+                  " = " + str(output1_data[0][i]))
+            if (input0_data[0][i] + input1_data[0][i]) != output0_data[0][i]:
+                print("sync infer error: incorrect sum")
+                EXIT_CODE_ = 1
+            if (input0_data[0][i] - input1_data[0][i]) != output1_data[0][i]:
+                print("sync infer error: incorrect difference")
+                EXIT_CODE_ = 1
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v',
+                        '--verbose',
+                        action="store_true",
+                        required=False,
+                        default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-t',
+                        '--concurrency',
+                        type=int,
+                        required=False,
+                        default=8,
+                        help='Request concurrency. Default is 8.')
+    parser.add_argument('-u',
+                        '--url',
+                        type=str,
+                        required=False,
+                        default='localhost:8000',
+                        help='Inference server URL. Default is localhost:8000.')
+    parser.add_argument(
+        '-i',
+        '--protocol',
+        type=str,
+        required=False,
+        default='http',
+        help='Protocol ("http"/"grpc") used to ' +
+        'communicate with inference service. Default is "http".')
+    FLAGS = parser.parse_args()
+    if (FLAGS.protocol != "http") and (FLAGS.protocol != "grpc"):
+        print("unexpected protocol \"{}\", expects \"http\" or \"grpc\"".format(
+            FLAGS.protocol))
+        exit(1)
+
+    threads = []
+
+    for idx in range(FLAGS.concurrency):
+        thread_name = "thread_{}".format(idx)
+
+        threads.append(
+            threading.Thread(target=infer,
+                             args=(FLAGS.protocol, FLAGS.url, FLAGS.verbose)))
+
+    for t in threads:
+        t.start()
+
+    for t in threads:
+        t.join()
+
+    sys.exit(EXIT_CODE_)
```

Comment:
Use `unittest` instead..

Reply:
or just reuse existing client

Reply:
Reused our existing clients. Removed this script.